### PR TITLE
GOVSI-813: Add new trust mark lambda + addition to well known handler

### PR DESCRIPTION
**Pull request by cj1993**
## What?

Add new Trust mark lambda returning static content + add to well known handler

### CODE REVIEW

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/WellknownHandler.java [hunk 1]
```diff
@@ -104,6 +105,14 @@ public APIGatewayProxyResponseEvent handleRequest(
                                 providerMetadata.setServiceDocsURI(new URI("http://TBA"));
                                 providerMetadata.setEndSessionEndpointURI(
                                         buildURI("/logout", baseUrl));
+                                providerMetadata
+                                        .toJSONObject()
+                                        .appendField(
+                                                "TrustMarkHandler",
```
Comment (sanisc) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/WellknownHandler.java:111]: so the key should be "trustmarks" and the value is the api endpoint for trustmarks you have just created

**Review by sanisc — CHANGES_REQUESTED** [review]

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/WellknownHandler.java [hunk 2]
```diff
@@ -104,6 +104,9 @@ public APIGatewayProxyResponseEvent handleRequest(
                                 providerMetadata.setServiceDocsURI(new URI("http://TBA"));
                                 providerMetadata.setEndSessionEndpointURI(
                                         buildURI("/logout", baseUrl));
+                                providerMetadata
+                                        .toJSONObject()
+                                        .appendField("trustmarks", "/trustmark");
```
Comment (sanisc) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/WellknownHandler.java:109]: you need a fully qualified name here for the url including the base url

**Review by sanisc — CHANGES_REQUESTED** [review]

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/WellknownHandler.java [hunk 3]
```diff
@@ -105,9 +106,19 @@ public APIGatewayProxyResponseEvent handleRequest(
                                 providerMetadata.setEndSessionEndpointURI(
                                         buildURI("/logout", baseUrl));
 
+                                providerMetadata =
+                                        OIDCProviderMetadata.parse(
```
Comment (JHjava) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/WellknownHandler.java:110]: Is it worth adding this an extra check to the WellknownHandlerTest to see if it is parsed correctly?

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/TrustMarkHandler.java [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+                            return generateApiGatewayProxyResponse(
+                                    400, OAuth2Error.INVALID_REQUEST.toJSONObject().toJSONString());
+                        });
+    }
+
+    private TrustMarkResponse createTrustMarkResponse() {
+        return new TrustMarkResponse(
+                configurationService.getBaseURL().orElseThrow(),
+                configurationService.getBaseURL().orElseThrow(),
+                List.of("Cl", "Cm", "Ch", "Cv"));
```
Comment (JHjava) [inline oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/TrustMarkHandler.java:51]: What does this list represent?

Comment (cj1993) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/TrustMarkHandler.java:51]: https://github.com/alphagov/digital-identity-architecture/pull/17/files

File: oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/WellknownHandler.java (hunk 3 above)
Comment (cj1993) [reply oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/WellknownHandler.java:110]: A ParseException will be thrown if the parse goes wrong, I think we will be fine with this

**Review by sanisc — APPROVED** [review]

**Pull request merged**